modules/op-adding-triggers.adoc

```diff
@@ -0,0 +1,151 @@
+// This module is included in the following assembly:
+//
+// *openshift_pipelines/op-creating-applications-with-cicd-pipelines.adoc
+
+
+[id="adding-triggers_{context}"]
+= Adding Triggers to a Pipeline
+
+[discrete]
+.Procedure
+After you have assembled and started the Pipeline for the application, add TriggerTemplates, TriggerBindings, and EventListener to capture GitHub events.
+
+. Copy the content of the following sample `TriggerBinding` YAML file and save it:
++
+----
+apiVersion: tekton.dev/v1alpha1
+kind: TriggerBinding
+metadata:
+  name: vote-app
+spec:
+  params:
+  - name: git-repo-url
+    value: $(body.repository.url)
+  - name: git-repo-name
+    value: $(body.repository.name)
+  - name: git-revision
+    value: $(body.head_commit.id)
+----
+
+. Create the `TriggerBinding`:
++
+----
+$ oc create -f <triggerbinding-yaml-file-name.yaml>
+----
++
+Alternatively, you can create `TriggerBinding` directly from the `pipelines-tutorial` Git repository:
++
+----
+$ oc create -f https://raw.githubusercontent.com/openshift/pipelines-tutorial/master/03_triggers/01_binding.yaml
+----
+
+. Copy the content of the following sample `TriggerTemplate` YAML file and save it:
++
+----
+apiVersion: tekton.dev/v1alpha1
+kind: TriggerTemplate
+metadata:
+  name: vote-app
+spec:
+  params:
+  - name: git-repo-url
+    description: The git repository url
+  - name: git-revision
+    description: The git revision
+    default: master
+  - name: git-repo-name
+    description: The name of the deployment to be created / patched
+
+  resourcetemplates:
+  - apiVersion: tekton.dev/v1alpha1
+    kind: PipelineResource
+    metadata:
+      name: $(params.git-repo-name)-git-repo-$(uid)
+    spec:
+      type: git
+      params:
+      - name: revision
+        value: $(params.git-revision)
+      - name: url
+        value: $(params.git-repo-url)
+
+  - apiVersion: tekton.dev/v1alpha1
+    kind: PipelineResource
+    metadata:
+      name: $(params.git-repo-name)-image-$(uid)
+    spec:
+      type: image
+      params:
+      - name: url
+        value: image-registry.openshift-image-registry.svc:5000/pipelines-tutorial/$(params.git-repo-name):latest
+
+  - apiVersion: tekton.dev/v1beta1
+    kind: PipelineRun
+    metadata:
+      name: build-deploy-$(params.git-repo-name)-$(uid)
+    spec:
+      serviceAccountName: pipeline
+      pipelineRef:
+        name: build-and-deploy
+      resources:
+      - name: git-repo
+        resourceRef:
+          name: $(params.git-repo-name)-git-repo-$(uid)
+      - name: image
+        resourceRef:
+          name: $(params.git-repo-name)-image-$(uid)
+      params:
+      - name: deployment-name
+        value: $(params.git-repo-name)
+----
```

Comment:
Would be great to have call outs here. Maybe for triggerbindings and event listener too, but this is a particularly big one. WDYT?

Reply:
As discussed on slack will not be adding callouts for now.